src/Microsoft.Extensions.DependencyInjection.Ordered/OrderedEnumerableServiceDescriptor.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection.Ordered
+{
+    public class OrderedEnumerableServiceDescriptor: ServiceDescriptor
```

Comment:
I played with this a little and I'm wondering what the benefit of adding `OrderedEnumerableServiceDescriptor`? Why not just store a list of service descriptors on the singleton itself `OrderedEnumerableServiceDescriptorContainer` itself?